src/app/routes/connector-ui/contract-agreement-page/contract-agreement-cards/contract-agreement-card-mapped.service.ts

```diff
@@ -43,4 +45,25 @@ export class ContractAgreementCardMappedService {
   ): ContractAgreementCardMapped[] {
     return search(cards, searchText, (card) => card.searchTargets);
   }
+
+  setStatus(
+    maxConsumingContracts: number,
+    agreements: ContractAgreementCardMapped[],
+  ): ContractAgreementCardMapped[] {
+    return agreements.map((it, index) => ({
+      ...it,
+      isConsumingLimitsEnforced: true,
+      statusText: index < maxConsumingContracts ? 'Active' : 'Inactive',
+      statusTooltipText:
+        index >= maxConsumingContracts
+          ? maxConsumingContracts == 1
+            ? 'This connector is configured with a limit of ' +
+              maxConsumingContracts +
+              ' active consuming contract agreement, causing contract agreement to get disabled if new ones are negotiated.'
+            : 'This connector is configured with a limit of ' +
+              maxConsumingContracts +
+              ' active consuming contract agreements, causing contract agreements to get disabled if new ones are negotiated.'
+          : '',
```

Comment:
let's add a method `getConsumingContractsInfoText(index: number, max: number): string` and use if-statements.